compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7606,6 +7608,21 @@ private BType checkMappingField(RecordLiteralNode.RecordField field, BType mappi
                             }
                         }
                     }
+                    if (!spreadRecordType.sealed) {
+                         if (mappingRecordType.sealed) {
+                             dlog.error(spreadExpr.pos,
+                                     DiagnosticErrorCode.INVALID_SPREAD_FIELD_TO_CREATE_CLOSED_RECORD_FROM_OPEN_RECORD,
+                                     spreadRecordType);
+                             errored = true;
+                         } else if (!types.isAssignable(spreadRecordType.restFieldType,
+                                 mappingRecordType.restFieldType)) {
+                             dlog.error(spreadExpr.pos,
+                                     DiagnosticErrorCode.INVALID_SPREAD_FIELD_REST_FIELD_MISMATCH,
+                                     mappingRecordType, mappingRecordType.restFieldType,
+                                     spreadRecordType.restFieldType);
```

Comment:
Are the arguments correct?

E.g.,

```ballerina
type Country record {|
    string name;
    string continent;
    string...;
|};

function testSpreadOpFieldOfMismatchingRestType() {
    record {string name; string continent; int population;} country1 = 
        {name: "China", continent: "Asia", population: 1400};
    Country _ = {...country1};
}
```

Current error is

```
invalid usage of spread field with open record of type 'Country', that may have rest fields of type 'string', to construct a record that allows only 'anydata' rest fields
```

Should be 

```
invalid usage of spread field with open record of type 'record {| string name; string continent; int population; anydata...; |}', that may have rest fields of type 'anydata', to construct a record that allows only 'string' rest fields
```

Reply:
Updated